Events show & index page + sponsor links

tochman opened the pull request:
closes #765

* Finishing touches on @yggie 's https://github.com/AgileVentures/WebsiteOne/pull/739 - `Events#show`
* Refactored `Events#index" view
    * Moved logic from view to helper
    * Changed layout 
    * Added `Join now` button and message for live events
* Added css classes to hide sponsor links on mobile devices
* Commented out the `puts` calls in `github_commits_job.rb`

**I refactored the features and view specs for the modified views but I left some scenarios in `hangout.feature` commented out. We should discuss if they should be removed.**

#### Screenshots 

(Don't mind the missing icon on one of them...)
![screenshot_2014-02-03-07-01-09 439](https://cloud.githubusercontent.com/assets/5248170/12974116/b0a3240a-d0b1-11e5-8c16-371471bdb740.png)
<img width="775" alt="skarmavbild 2016-02-10 kl 20 44 15" src="https://cloud.githubusercontent.com/assets/5248170/12974117/b0aa1a58-d0b1-11e5-9633-14c9509dda1c.png">

tansaku commented:
not sure who added "closes #750, closes #749", but running this locally I still get a fair amount of junk in the rspec and cucumber logs, so while it might be improving things I don't know if we can close those two issues with this pull request, although perhaps we should split out separate issues for individual junk items ...

tochman commented:
I added that. I am not getting junk anymore...

tansaku commented:
This all passes for me locally @tochman with the exception of:

```
(::) failed steps (::)

Unable to find css "#liveHOA-placeholder iframe" (Capybara::ElementNotFound)
./features/step_definitions/hangout_steps.rb:2:in `/^I should see hangout button$/'
features/hangout.feature:103:in `Then I should see hangout button'

Failing Scenarios:
cucumber features/hangout.feature:100 # Scenario: Display hangout button on a project's page
```

but I'm seeing that failure quite a lot locally

tochman commented:
yes, that is passing on CI. I think #710 addressed that issue but was never finished.

tansaku commented:
junk that I still see in the test output:

```
[tansaku@Samuels-MBP-2:~/Documents/Github/AgileVentures/WebSiteOne ((6bd54e4...))]$ 
→ ber

Randomized with seed 7353
............................................................................................................................................../Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/app/models/newsletter.rb:15: warning: previous definition of CHUNK_SIZE was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: previous definition of CHUNK_SIZE was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: previous definition of CHUNK_SIZE was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: previous definition of CHUNK_SIZE was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: previous definition of CHUNK_SIZE was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:14: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:7: warning: previous definition of CHUNK_SIZE was here
.The user session path is /users/sign_in
.The user session path is /users/sign_in
.The user session path is /users/sign_in
.The user session path is /users/sign_in
.The user session path is /users/sign_in
.........An expectation of :id was set on nil. Called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/support/shared_examples/shared_example_for_hangout_button.rb:6:in `block (2 levels) in <top (required)>'. Use allow_message_expectations_on_nil to disable warnings.
.An expectation of :id was set on nil. Called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/support/shared_examples/shared_example_for_hangout_button.rb:6:in `block (2 levels) in <top (required)>'. Use allow_message_expectations_on_nil to disable warnings.
..............*.....................................................................................................................................................................................................................................-- add_column(:events, :start_time, :time)
   -> 0.0047s
-- add_column(:events, :event_date, :date)
   -> 0.0003s
-- add_column(:events, :end_time, :time)
   -> 0.0004s
-- remove_column(:events, :start_datetime, :datetime)
   -> 0.0006s
-- remove_column(:events, :duration, :integer)
   -> 0.0003s
.-- add_column(:events, :start_time, :time)
   -> 0.0006s
-- add_column(:events, :event_date, :date)
   -> 0.0005s
-- add_column(:events, :end_time, :time)
   -> 0.0003s
-- remove_column(:events, :start_datetime, :datetime)
   -> 0.0004s
-- remove_column(:events, :duration, :integer)
   -> 0.0004s
-- add_column(:events, :start_datetime, :datetime)
   -> 0.0003s
-- add_index(:events, :start_datetime)
   -> 0.0019s
-- add_column(:events, :duration, :integer)
   -> 0.0006s
-- remove_column(:events, :start_time, :time)
   -> 0.0004s
-- remove_column(:events, :event_date, :date)
   -> 0.0003s
-- remove_column(:events, :end_time, :time)
   -> 0.0003s
............................................................................................................................An expectation of :includes was set on nil. Called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/controllers/projects_controller_spec.rb:46:in `block (3 levels) in <top (required)>'. Use allow_message_expectations_on_nil to disable warnings.
............................................................./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/app/models/newsletter.rb:14: warning: previous definition of SEND_AS was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: previous definition of SEND_AS was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: previous definition of SEND_AS was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:46: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/jobs/send_newsletter_spec.rb:14: warning: previous definition of CHUNK_SIZE was here
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:47: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:42: warning: previous definition of SEND_AS was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:24: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:46: warning: previous definition of CHUNK_SIZE was here
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:25: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:47: warning: previous definition of SEND_AS was here
./Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:24: warning: already initialized constant Newsletter::CHUNK_SIZE
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:24: warning: previous definition of CHUNK_SIZE was here
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:25: warning: already initialized constant Newsletter::SEND_AS
/Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/models/newsletter_spec.rb:25: warning: previous definition of SEND_AS was here
..................................................................................invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
....................................................................................................................................................

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) events/new user is not signed in should redirect to sign in and render a flash error
     # Temporarily skipped with xit
     # ./spec/views/events/new.html.erb_spec.rb:18


Deprecation Warnings:

Using `should` from rspec-expectations' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` with `config.expect_with(:rspec) { |c| c.syntax = :should }` instead. Called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/controllers/documents_controller_spec.rb:118:in `block (3 levels) in <top (required)>'.

Using `stub` from rspec-mocks' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` instead. Called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/spec/controllers/documents_controller_spec.rb:22:in `block (2 levels) in <top (required)>'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.

2 deprecation warnings total

Finished in 1 minute 13.55 seconds (files took 9.37 seconds to load)
827 examples, 0 failures, 1 pending

Randomized with seed 7353

Coverage report generated for RSpec to /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/coverage. 1448 / 1551 LOC (93.36%) covered.
[Coveralls] Outside the CI environment, not sending data.
[tansaku@Samuels-MBP-2:~/Documents/Github/AgileVentures/WebSiteOne ((6bd54e4...))]$ 
→ bec
[Coveralls] Using SimpleCov's 'rails' settings.
Using the default profile...
...............................................................................................................................................................DEPRECATION WARNING: Passing string argument to #within_window is deprecated. Pass window object or lambda. (called from /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/features/step_definitions/articles_steps.rb:24)
..............................................................................................................invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
...............................invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.....................................................................................................................................................................................................................................................................................................................F    HTML screenshot: /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/tmp/capybara/screenshot_2014-02-03-07-01-06.710.html
    Image screenshot: /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/tmp/capybara/screenshot_2014-02-03-07-01-06.710.png
..................................................invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
..........invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
...................................................................................................................................................................................................................................................................................................................-----............................................................................................................................................................................................
```

the 'something is wroonnnngggg!" thing is gone, but there is quite a lot of other non-test specific output that it would be great to clean out

tansaku commented:
btw @tochman great to be working with you closely on a coding project again - it's been a couple of years I guess :-)

tochman commented:
About `junk` perhaps we mean different things. I was under the impression that `junk` referred to `Somethings wrooong`etc in the test output.

tansaku commented:
that's junk, but there's other junk too right? e.g. 

```
invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
.invalid keys :visible, should be one of :text, :visible, :between, :count, :maximum, :minimum, :exact, :match, :wait
```

tansaku commented:
with `-fp` or `--format` progress we should only see dots :-)

from my .bashrc

my .bashrc

```
alias ber="bundle exec rspec -fp"
alias bec="bundle exec cucumber --format progress"
```

tochman commented:
Okay. I get the point. Should I remove the `close` or should I remove the junk?  Hmm... ;-) Please advice @tansaku

tansaku commented:
not saying that we need to fix all that in this pull request. 

we could adjust those issues to specify the "something is wrong" thing specifically and then open other issues for the other bits of junk - then we get a nice satisfying burn down :-)

tochman commented:
It's your call. I feel a little lost here... :-( The big question is if the new UI is okay and we can move forward with this PR.

tansaku commented:
yes, don't worry about the junk/issues - I'll handle that.  At least we've worked out how we are using the word "junk" :-)

so I'm running this locally and the event show page now looks like this:

![event show page](https://www.dropbox.com/s/6x7iigaaq7ggu5l/Screenshot%202016-02-11%2017.09.15.png?dl=1)

Trying to edit I got a mutex lock, but hopefully that's just local to me.

Main thing I notice is that we've lost the ability to edit the hangout link.  That manual override has been a critical functionality for me when the hangout connection fails.  I'd really prefer not to lose that functionality, or is it now available somewhere else?

![current hangout link edit functionality](https://www.dropbox.com/s/yz5hg88pe77ag9q/Screenshot%202016-02-11%2017.12.16.png?dl=1)

tansaku commented:
btw @tochman great to be working with you closely on a coding project again - it's been a couple of years I guess :-)

Overall the page looks a lot cleaner, but I'd like to change mention of "scrums" to "standups".

Some other thoughts from using it.

I think it's really confusing that when you finish editing an event you end up on the events list page ... (but that's how it's always been.)

Also the list page now only has one column ... which I like - would be great if all the dates were on a single line to reduce the space taken by each event on each row, but none of those are blockers in principle.  Only exception in my mind is losing the ability to edit the hangout link ...

tochman commented:
@tansaku Yeah, good stuff being back on this project - if only for a brief period ;-) 

Okay, so the spec for this feature was to remove the edit hangout link functionality the way i remember it (That's probably it was gone on @yggie 's draft). But I see your point.

tochman commented:
Roger on `"scrums" to "standups"`

tansaku commented:
interesting - maybe that should be something only for admins - I need to review the original spec.

Also haven't had time to review tests or code - will get to that tomorrow ...

tochman commented:
I reintroduced the edit Hangout link. Here is what happens:
* Editing the link option is in the Event Actions pulldown (viewable only for signed in users)
* When we click `Edit hangout link` the form is revealed. However, the pulldown menu remains open (pulled down). **I can probably make it close it with jQuery but the question is if it's worth the hustle?**
* If there is a hoa link on the `recent_hangout` object, that link will populate the form field. If not, there is a placeholder text in the input field.
* If a link is added manually the Event automatically goes Live and the 'Join now' link is added. 

### The Missing Step

```gherkin
And I should see "Event URL has been updated"
```

The problem that I see with this atm is that we are not giving the user (updater) any feedback. Back in the days, when me and @yggie implemented the first version if this feature, we had a method in the `events_controller` that, with a successful update rendered a flash. After several refactorings, that method was apparently not being used (I deleted it in this PR) and the update is done on the `EventInstance#update` - perfectly logical. It is a quite messy update method that I don't felt like touching so my question is: **How important is the flash feedback in the grand scale of things?** Please advice @tansaku  



![skarmavbild 2016-02-12 kl 08 21 30](https://cloud.githubusercontent.com/assets/5248170/13001016/b8c9d42a-d161-11e5-9996-91691df9ca95.png)

tansaku commented:
flash feedback not required here I would say @tochman - this is just what the doctor ordered!

tansaku commented:
Also since one can re-edit to see if the link has been updated - the flash is not so important.

We should probably also have a conversation about the scope of pull requests :-)

tochman commented:
Haha! Yes. Of course. I'm a little rusty @tansaku Sorry about that. Will strive to get back in line at live up to the standards that I was part of implementing.

tansaku commented:
no worries :-) 

So when I merge in the fixes I did in the other branch to upgrade capybara, poltergeist and fix phantomjs this branch is all green locally for me ...

we should talk about the commented out tests ... those tests relate to editing the hangout link and restarting the hangout (another feature we lost right ...?)

@sampritipanda also makes a good point about moving the puts statements to logger statements, but I'm also keenly aware that you don't have much free time, so don't want to load requirements onto *this* pull request - some things we can merge in and make chores for subsequent changes ...

I'll have a quick go at fixing some of these tests ...

tansaku commented:
I got a quick pass of one of them:

```
  @javascript
  Scenario: Create a hangout for a scrum event
    Given I am on the show page for event "Scrum"
    Then I should see hangout button
    And I click "Event Actions"
    And I should see "Edit hangout link"
```

although that test is poorly named and redundant and could safely be deleted if the "edit hangout link" is used in another test, which I think it is.

Looking at:

```
  Scenario: Show hangout details
    Given the Hangout for event "Scrum" has been started with details:
      | EventInstance link | http://hangout.test |
      | Started at         | 10:25:00 UTC        |
    And the time now is "10:26:00 UTC"
    When I am on the show page for event "Scrum"
    Then I should see Hangouts details section
    And I should see:
      | Category            |
      | Scrum               |
      | Daily scrum meeting |
      | Updated             |
      | 1 minute ago        |
    And I should see link "http://hangout.test" with "http://hangout.test"
```

I removed "title" and got onto this not finding "updated", which I think is something we need to let us know when hangout links are old ones versus new ones ...

tansaku commented:
I went on to try and get the "Cancel Edit URL" scenario working but hit an issue

```
  @javascript
  Scenario: Cancel Edit URL
    Given I am on the show page for event "Scrum"
    And I click "Event Actions"
    When I click the link "Edit hangout link"
    And I click "Cancel"
    Then I should not see button "Save"

```

```
  @javascript
  Scenario: Cancel Edit URL                       # features/hangout.feature:79
    Given I am on the show page for event "Scrum" # features/step_definitions/event_steps.rb:60
    And I click "Event Actions"                   # features/step_definitions/basic_steps.rb:73
    When I click the link "Edit hangout link"     # features/step_definitions/basic_steps.rb:93
    And I click "Cancel"                          # features/step_definitions/basic_steps.rb:73
      Firing a click at co-ordinates [746, 451] failed. Poltergeist detected another element with CSS selector 'html body div#wrap div#main div.container div.col-lg-9 div.row div.col-xs-6 div.dropdown.dropdown-beside-HOA.pull-right.open div.dropdown-backdrop' at this position. It may be overlapping the element you are trying to interact with. If you don't care about overlapping elements, try using node.trigger('click'). (Capybara::Poltergeist::MouseEventFailed)
      ./features/step_definitions/basic_steps.rb:74:in `/^(?:when I|I) click "([^"]*)"$/'
      features/hangout.feature:83:in `And I click "Cancel"'
    Then I should not see button "Save"           # features/step_definitions/basic_steps.rb:215
capybara-screenshot could not detect a screenshot driver for 'poltergeist_ignore_ssl'. Saving with default with unknown results.
    HTML screenshot: /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/tmp/capybara/screenshot_2016-02-12-13-15-04.947.html
    Image screenshot: /Users/tansaku/Documents/Github/AgileVentures/WebSiteOne/tmp/capybara/screenshot_2016-02-12-13-15-04.947.png

Failing Scenarios:
cucumber features/hangout.feature:79 # Scenario: Cancel Edit URL
```

and I'm going to go to lunch 

I note that if this branch was on the websiteone repo rather than your own we could be asynchronously ping pong pairing on it :-)

don't feel any pressure to work on this, I know how busy you are, but if you are going to do a few things - consider moving it onto website one and doing the pull request from there.  I won't do anything more substantial before Monday, but can do some more next week.

Either way don't feel under pressure to throw yourself at this. I really appreciate you stepping in to finish up @yggie 's thing, but please don't stress yourself taking this the last mile :-)

tochman commented:
@tansaku I've never (or almost nerver) worked against the main repository and I'm not convinced that is a good idea. Unless you really want me to, I would suggest that you create a PR to my feature branch and I take it from there.

tansaku commented:
when I say work against the main repo, I mean branches on the main repo - I do it all the time on LocalSupport - did it all the time at Makers for 18 months and it is awesome.

Can you point out one conceivable downside of working on a branch on the main repo?

tochman commented:
Nothing serious apart from limiting other members from participating in the process.

tochman commented:
I have no clue how to test the collapsed in Edit link form when the `Cancel` button has been clicked. I got a few more cucumber scenarios to pass but I'm stuck on this one. 
```
 @javascript
  Scenario: Cancel Edit Hangout URL
    Given I am on the show page for event "Scrum"
    And I click "Event Actions"
    And I click the "Edit hangout link" link
    And I click on the Cancel button
    Then I should not see the Edit URL controls
```
I've tried with different approaches in `basic_steps.rb:85+`. Pushing it up as WIP to main AV repository.

tochman commented:
@tansaku I added a js method to hide the dropdown on menu item click and toggle the menu item text. But I'm blocked on testing that in Jasmine. I added two specs to `evens_spec.js`  but I have problem with what fixture the test is loading so they will not pass. Also, I notice that those specs are testing functionality we are no longer using. 
![av_events](https://cloud.githubusercontent.com/assets/5248170/13027196/a2fe54d8-d245-11e5-950f-1f149a4e5d97.gif)

tansaku commented:
Thanks for updating this @tochman - I've had a look at the jasmine tests and I can see there's a lot to review there (like a complete overhaul of the fixtures etc.).  I've been looking at the tests in the individual test runner using `rake jasmine`,  and wondering if we really need these jasmine tests?  Isn't the functionality there being tested by the acceptance tests already.

I'd take a deeper look, but I'm about to go to lunch.   Will look again later today. 

I'm also a little concerned about feature creep in this pull request.  It's now got some additional components changing things related to sponsors etc.  Also removing that github related log junk is taken care off with a rails logger in another pull request.

If the sponsor stuff was in a separate pull request we'd be able to get that into production faster without it being held up by the complexities of this pull request.  Perhaps we can split those things out?

I don't want to load work on to you so perhaps we can cherry pick commits out and take then into other pull requests ourselves?

Also if we have failing tests maybe better to leave them failing rather than commented out?  That way the CI points the maintainers to the location of the problem rather than comments in the code?  Just an idea.

Looking forward to chatting about the workflow in real time at some point.

tochman commented:
The failing cucumber scenarios that are commented out atm are testing a feature that is no longer in place. I left them there for review purposes. Will remove them now.

You are right about the sponsorship fix. It shouldn't be in this PR. I would let this one slip though. It is a small change and the work related to pulling this out seems to me as too much work for what we actually get. (This relates only to the mobile usage of the site) But I leave the decision up to you.

About the Jasmine test. I have no opinion about if we should keep Jasmine. I agree with your assessment that most of this is tested by acceptance tests. But this is a bigger issue, also outside the scope of this PR.

tansaku commented:
I meant the commented out jasmine tests :-) I'll explain in real time if we can grab a moment later in the week.

I'm also a little shy of removing the 'restart hangout' feature which has been an important admin tool on a few occasions.

If you're okay to leave it with me I will cherry-pick this PR apart into a series of smaller PRs so we merge separately ...

tochman commented:
So the 'restart hangout' was deemed unnecessary and was to be removed in this redesign according to the original specs. We get almost the same functionality with the Edit hangout URL. The whole point with redesigning this was to make the page much simpler and strip it of unused and complex functionality.

If you want to spend time splitting this up you are welcome to do so. I don't see the need for that to be honest. I understand the removal of the jasmine specs if we are not going to work on that but the rest seems like a unnecessary burden. :smile:

tochman commented:
So this is the original wireframe that we agreed on for this story (it was actually a 2 point story) 
https://docs.google.com/file/d/0B9FqKjpP8_MUSnhCYmpKMi1URFk/edit

tansaku commented:
thanks @tochman - splitting up might be a burden if you needed to do it - but I believe that it will make things much easier for me in terms of learning about the ins and outs of the project.

We've just split out the critical parts of the github commit count pull request which was a lot of git work, but it means that in #800 it's a lot easier to see what's going on.  I plan to do the same here (tomorrow if I have time), and I hope you'll see the benefit in the long run :-)

tochman commented:
I'd love to see the results of that process and what benefits that brings. I might be able to learn something from that apart from the lessons drawn already (that I should have stayed in scope of the feature) The Sponsor links visibility on mobile devices are part of db42ad1c48c80b09c8d202c16c8b298d88ffa4db

dalupus commented on app/views/events/show.html.erb:
```diff
@@ -1,54 +1,116 @@
+        <i class="fa fa-calendar"></i>
+
+        <div class="calendar-text">
+          <%= @event.next_event_occurrence_with_time[:time].strftime('%A, %B %d, %Y') %><br/>
```
We should local_time this to be consistent.

diraulo commented:
It says `started` and `ended` on events that are yet to take place.

<img width="428" alt="screenshot 2016-02-17 19 42 11" src="https://cloud.githubusercontent.com/assets/4028374/13118991/8618866e-d5af-11e5-8cd8-11664a8dc814.png">

Perhaps we should just keep the start and end time as depicted on the wireframe?
<img width="207" alt="screenshot 2016-02-17 19 51 08" src="https://cloud.githubusercontent.com/assets/4028374/13119054/d34066be-d5af-11e5-996a-6fb3f1163c74.png">

tansaku commented:
@diraulo: this all seems to work for me locally (tests are green) - although I do need `phantomjs: Phantomjs.path` in the Capybara test options since I uninstalled phantomjs globally, but I can put a separate pull request through for that ... I can't immediately see how I can check that the event page updates correctly with a live hangout, but I guess we can test that on develop?  and we don't have lots of other pull requests that we'll blocking on so if you're happy to merge this I'd say go ahead and we'll fix any issues up retrospectively (like deleting view specs etc.)  since this is kind of a legacy pull request ...

so please go ahead and merge when you are ready ...

diraulo commented:
Sure thing @tansaku :+1:

tansaku commented:
@diraulo oh sorry, just saw that comment about "started - ended" - I agree, but I think we don't want to load more onto @tochman to work on here - let's fix that up ourselves before we push to production

tochman commented:
It does NOT say `started` and `ended` on events that are yet to take place.
```
# features/events.feature:14
 Scenario: Show index of events
    Given the date is "2014/02/01 09:15:00 UTC"
    And I am on Events index page
    Then I should see "AgileVentures Events"
    And I should see "Scrum"
    And I should see "Starts at 07:00 - Ends at 09:30 (UTC)"
    And I should see "PP Session"
    And I should see "Starts at 10:00 - Ends at 10:15 (UTC)"
```
Or am I missing something?

diraulo commented:
I don't know @tochman. If you have a look at that screenshot. The event is set for the 18th of Feb (SAST) ... It's still the 17th in my part of the world :-)

tochman commented:
I bet it is @diraulo ;-) Well, I need to have a look at this... Weird...

tochman commented:
@diraulo @tansaku I messed up the helper methods for events trying to compare time in different formats. Pushing up a refactored method

dalupus commented:
On the show page I am ok with moving local time to a future PR.  But I would like to atleast have the timezone stated in this one.   The user has no indication that the timezone has suddenly switched to utc I could be quite confusing to someone who does not realize that.  Simply hardcoding UTC after the time is fine for now.

tochman commented:
Closed by mistake

tochman commented:
The way I see it, the live? method on EventInstance is problematic. 

def live?
    started? && hoa_status != 'finished' && updated_at > 2.minutes.ago
end

The way it is now, the event will be live in two minutes only, after that the 'Join now' link will disappear from both #index and #show pages. 

Why 2 minutes? Thoughts on that @tansaku ?

tansaku commented:
you mean live for two minutes only?  that doesn't sound good.  I thought we had it 15 mins by default, although @alejandrobabio had been looking into making the whole thing work so that it was actually live when the broadcast was live but I don't know how far he got with that ... I know he had some successes, but it wasn't clear to me how far he got that functionality through the codebase ...

hmm, if we've still got work to do here, I might think to merge #800 first and fix up the commit conflicts here rather than the other way around that was my original plan ...

@tochman I know that your preference is to have the PR judged based on the specifics of the story that it's meant to address, but with such a key feature of our site, perhaps it bears careful reflection ...

tochman commented:
So it seems that the 2 minutes is the interval in which the HOA connection is talking back to WSO. The question we might want ask ourselves is how that affects performance but that is outside of this story.

tochman commented:
My last comment means that I don't see the `live?` method as a blocker.

alejandrobabio commented:
About the 2 minutes issue: when a hangout session starts, then each 2 minutes the hangout app sends a message to agileventures, with the current hangout status. It could be 'started', 'broadcasting' or 'finished'. 

The 'started' state means that the hangout page is open, but it is not live yet. If you wait 10 minutes before change to 'live' (or click 'Start broadcasting' button), then it sends 5 messages with 'started' status to agileventures.
When the hangout becomes live, the status become 'broadcasting'. The messages keep going each 2 minutes with the new state.
An if you end the broadcasting with the 'Stop Broadcast' button, it sends a last message to AV with hoa_status 'finished'. But if you just close the hangout window, this last message is never sent. And that's why the 2 minutes delay before respond to `live?` with `false`.

I did test it at https://www.youtube.com/watch?v=Aq7O0WuGSLI

This `live?` state at `EventInstance` model will help us to show the "right" state of the event on the landing page (yes, this is pending yet).

I hope it helps you.

tochman commented:
Thank you for the clarification Alejandro. 

On another note, we still have a problem with the time range helpers. They work as expected on new instances of EventInstance but fail on existing ones. Will write up a ticket for this.
